### PR TITLE
support configuration-based relative dirs (host and agent) for iofs

### PR DESCRIPTION
**Pull request by zootalures**
I noticed  that http-stream doesn't work on docker4mac - this is because docker mounts volumes relative to the OSX host, not the docker VM. 

This change allows two directory roots for IOFS mounts one with respect to the docker host, one with respect to the agent itself.  The behaviour is conditional on both FN_IOFS_PATH and FN_IOFS_MOUNT_ROOT. 

I've also added the specified headers to the contract (proper HTTP versions rather than the old FN_DEADLINE) as dups , this makes java happy

```
docker run -v $(pwd)/maciofs:/iofs -eFN_LOG_LEVEL=debug  -e FN_IOFS_PATH=/iofs -e FN_IOFS_MOUNT_ROOT=$(pwd)/maciofs  --privileged -v /var/run/docker.sock:/var/run/docker.sock   -p8080:8080  --name fnserver --rm  fnproject/fnserver
```

### CODE REVIEW

**Comment by zootalures** [comment]
Plan would be for `fn start` to use this behavior along side a new "don't create tmpfs mounts" option which would also (i think) allow us to run Fn unprivileged and make docker4windows work.

File: api/agent/agent.go [hunk 1]
```diff
@@ -726,18 +727,40 @@ func (s *hotSlot) dispatch(ctx context.Context, call *call) chan error {
...
+	// TODO(occ) : fix compatidupes when FDKs are updated
+	req.Header.Set("Fn-Call-Id", call.ID)
 	req.Header.Set("FN_CALL_ID", call.ID)
+	deadline, ok := ctx.Deadline()
+	if ok {
+		deadlineStr := deadline.Format(time.RFC3339)
+		req.Header.Set("Fn-Deadline", deadlineStr)
+		req.Header.Set("FN_DEADLINE", deadlineStr)
+
+	}
```
Comment (rdallman) [inline api/agent/agent.go]: thanks for ^ mods

**Comment by rdallman** [comment]
needs gofmt but LGTM

**Comment by zootalures** [comment]
Yep, oops will fix tomorrow

**Pull request merged**